src/schema/messages.json

```diff
@@ -0,0 +1,51 @@
+{
+  "id": "messages",
+  "types": {
+    "i18nPlaceholder": {
+      "type": "object",
+      "properties": {
+        "content": {
+          "type": "string"
+        },
+        "example": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "content"
+      ]
+    },
+    "i18nMessage": {
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "placeholders": {
+          "type": "object",
+          "patternProperties": {
+            "^[A-Za-z0-9_@]+$": {
+              "$ref": "#/types/i18nPlaceholder"
+            }
+          },
+          "additionalProperties": false
+        }
+      },
+      "required": [
+        "message"
+      ]
+    },
+    "WebExtensionMessages": {
+      "type": "object",
+      "patternProperties": {
+        "^[A-Za-z0-9_@]+$": {
+          "$ref": "#/types/i18nMessage"
+        }
+      },
+      "additionalProperties": false
+    }
+  }
+}
```

Comment:
Great idea!

cc @mstriemer on this as our "hero of schema" - I also wasn't able to find a schema for these messages files in mozilla central (neither https://hg.mozilla.org/mozilla-central/file/tip/toolkit/components/extensions/schemas/ nor https://hg.mozilla.org/mozilla-central/file/tip/browser/components/extensions/schemas) - maybe this is something we should push to upstream?

Reply:
As I noted in the pull request, the format of message names is not currently enforced in Firefox, I haven't checked in Chrome, but I assume that's where the limitation comes from. I've checked the source, and Firefox does enforce placeholder names: https://dxr.mozilla.org/mozilla-central/source/toolkit/components/extensions/ExtensionCommon.jsm#1672

It also seems to replace missing placeholders with an empty string, probably something worth noting somewhere, but not necessarily in the linter (but the page it links to)